File: plugins/yaml/src/org/jetbrains/yaml/YAMLFileElementType.java
// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package org.jetbrains.yaml;

import com.intellij.psi.tree.IStubFileElementType;

public class YAMLFileElementType extends IStubFileElementType {
  public YAMLFileElementType() {
    super(YAMLLanguage.INSTANCE);
  }

  @Override
  public int getStubVersion() {
    return 1;
  }
}

File: plugins/yaml/src/YamlFileElementType.kt
// Copyright 2000-2026 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package org.jetbrains.yaml

import com.intellij.platform.syntax.psi.SyntaxFileElementType
import com.intellij.psi.tree.IFileElementType

@JvmField
val YAML_FILE: IFileElementType = SyntaxFileElementType(YAMLLanguage.INSTANCE)